IDL/cocoshow.pro

```diff
@@ -42,27 +43,54 @@ function COCOSHOW, coco_data_rgb_int, quiet=quiet, current=current, dims=dims, f
 ; MODIFICATION HISTORY:
 ; 	Written by:	Malcolm Druett, May 2019
 ;-
-  if (n_elements(dims) eq 2) then begin   
-     nx=dims[0]
-     ny=dims[1]
-  endif else begin
-     sz=size(coco_data_rgb_int)
-     nx=sz[1]
-     ny=sz[2]
-  endelse
-  if keyword_set(quiet) then begin
-     w=WINDOW(DIMENSIONS=[nx,ny],/BUFFER) 
-  endif else begin
-     if (not keyword_set(current)) then begin
-        w=WINDOW(DIMENSIONS=[nx,ny])
-     endif
-  endelse
-  temp_image=image(coco_data_rgb_int, IMAGE_DIMENSIONS=[nx,ny], POSITION=[0.0,0.0,1.0,1.0],/current)
-  the_image=temp_image.CopyWindow()
+
+; displaying image
+  ; Set up options for correct display
+  if (keyword_set(quiet)) then option1=1 else option1=0
+  if (keyword_set(name)) then option2=1 else option2=-1
+  option=option1*option2
+  case option of
+     0: begin
+     ; not quiet, therefore either display in a new or 
+     ; the current window without /buffer option.
+        if (n_elements(dims) eq 2) then begin   
+           nx=dims[0]
+           ny=dims[1]
+        endif else begin
+           sz=size(coco_data_rgb_int)
+           nx=sz[1]
+           ny=sz[2]
+        endelse
+        if (not keyword_set(current)) then begin
+           w=WINDOW(DIMENSIONS=[nx,ny])
+        endif
+        temp_image=image(coco_data_rgb_int, IMAGE_DIMENSIONS=[nx,ny], POSITION=[0.0,0.0,1.0,1.0],/current)
+     end
+     1: begin
+     ; quiet, but save required, therefore display in buffer
+        if (n_elements(dims) eq 2) then begin   
+           nx=dims[0]
+           ny=dims[1]
+        endif else begin
+           sz=size(coco_data_rgb_int)
+           nx=sz[1]
+           ny=sz[2]
+        endelse
+        if (not keyword_set(current)) then begin
+           w=WINDOW(DIMENSIONS=[nx,ny],/BUFFER)
+        endif
+        temp_image=image(coco_data_rgb_int, IMAGE_DIMENSIONS=[nx,ny], POSITION=[0.0,0.0,1.0,1.0],/current)
+     end
+     ; no save or display required, so no image

```

Comment:
Add ` else: return` to exit the case statement in case `quiet` is set and no `name` supplied.

However, setting of `/BUFFER` in `WINDOW()` is the only difference between cases 0 and 1, so probably better to remove the case-check and instead go for 
```
if (not keyword_set(current)) then w = WINDOW(DIMENSIONS=[nx,ny], BUFFER=option)
```

Reply:
Removed case in latest commit.